lib/models/ratio.dart

```diff
@@ -0,0 +1,489 @@
+
+      case Ratio.milliLitersPerKiloGram:
+        return MASS.kilograms;
+
+      case Ratio.litersPerKilogram:
+        return MASS.kilograms;
+
+      case Ratio.kiloCaloriesPerOunce:
+        return MASS.ounces;
+
+      case Ratio.gramsPerSquareMeter:
+        return AREA.squareMeters;
+
+      case Ratio.kiloGramsPerSquareMeter:
+        return AREA.squareMeters;
+
+      case Ratio.microGramsPerSquareMeter:
+        return AREA.squareMeters;
+
+      case Ratio.milliGramsPerSquareMeter:
+        return AREA.squareMeters;
+
+      case Ratio.nanoGramsPerSquareMeter:
+        return AREA.squareMeters;
+
+      case Ratio.milliGramsPerDeciLiter:
+        return VOLUME.deciliters;
+
+      case Ratio.gramsPerDeciLiter:
+        return VOLUME.deciliters;
+
+      case Ratio.nanoGramsPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.picoGramsPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.microGramsPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.milliGramsPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.gramsPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.kiloGramsPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.milliGramsPerCubicMeter:
+        return VOLUME.cubicMeters;
+
+      case Ratio.kiloGramsPerCubicMeter:
+        return VOLUME.cubicMeters;
+
+      case Ratio.femtoMolesPerMilliLiter:
+        return VOLUME.microliters;
+
+      case Ratio.picoMolesPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.nanoMolesPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.microMolesPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.molesPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.picoMolesPerDeciLiter:
+        return VOLUME.deciliters;
+
+      case Ratio.nanoMolesPerDeciLiter:
+        return VOLUME.deciliters;
+
+      case Ratio.microMolesPerDeciLiter:
+        return VOLUME.deciliters;
+
+      case Ratio.milliMolesPerDeciLiter:
+        return VOLUME.deciliters;
+
+      case Ratio.milliMolesPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.picoMolesPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.nanoMolesPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.microMolesPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.molesPerLiter:
+        return VOLUME.liters;
+
+      case Ratio.molesPerCubicMeter:
+        return VOLUME.cubicMeters;
+
+      case Ratio.gramsPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.nanoGramsPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.picoGramsPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.microGramsPerDeciLiter:
+        return VOLUME.deciliters;
+
+      case Ratio.microGramsPerMilliLiter:
+        return VOLUME.milliliters;
+
+      case Ratio.milliGramsPerMilliliter:
+        return VOLUME.milliliters;
+    }
+  }
+}
```

Comment:
What's the point with Ratio? You defined it in the model folder even if it is not a model. Just an enum with 2 extensions. Would you like to create a new property?

Reply:
So for this one I wasn't sure the best way to approach it, or where it should go. 

My thought behind it was that there are lots of measurements that aren't basic or primitive measurements (length, mass, etc). But are ratios or combinations of measurements (g/mL, grams/mole, m/s^2 etc). 

I noted that you already had torque as a class which does combine two different measurements into one class. Obviously you could have infinite number of combining units, but even just using the more common ones there quite a few. Now, most of the ones I created could be grouped into just a few different groups (mass/volume, mass/mass, mass/area, amount of substance/mass).

And if you think this is the cleaner way to do it, I can work on that. But my thought was with so many ratios and combinations out there, if you could define a list of the measurements, it's not hard to break them down into their components, convert all of their components, and then return that value. Which I thought was a simpler way of doing it, especially as you started to get more complicated measurements (like the m/s^2).

Reply:
Seems reasonable, give me time to think about this. I'm sorry but I have just few hours in the weekends to dedicate to side project. The rest look good to me!